web/src/lib/components/user-settings-page/PincodeInput.svelte

```diff
@@ -0,0 +1,111 @@
+<script lang="ts">
+  interface Props {
+    label: string;
+    value?: string;
+    pinLength?: number;
+    tabindexStart?: number;
+  }
+
+  let { label, value = $bindable(''), pinLength = 6, tabindexStart = 0 }: Props = $props();
+
+  let pinValues = $state(Array.from({ length: pinLength }).fill(''));
+  let pincodeInputElements: HTMLInputElement[] = $state([]);
+
+  export function reset() {
+    pinValues = Array.from({ length: pinLength }).fill('');
+    value = '';
+  }
+
+  const focusNext = (index: number) => {
+    if (index < pinLength - 1) {
+      pincodeInputElements[index + 1]?.focus();
+    }
+  };
+
+  const focusPrev = (index: number) => {
+    if (index > 0) {
+      pincodeInputElements[index - 1]?.focus();
+    }
+  };
+
+  const handleInput = (event: Event, index: number) => {
+    const target = event.target as HTMLInputElement;
+    let currentPinValue = target.value;
+
+    if (target.value.length > 1) {
+      currentPinValue = value.slice(0, 1);
+    }
+
+    if (!/^\d*$/.test(value)) {
```

Comment:
`Number.isInteger()`?

Reply:
Not the same 

<img width="301" alt="image" src="https://github.com/user-attachments/assets/953112fb-9a63-4311-ba1b-2568b04656a1" />

Reply:
Ah sorry. You could do `Number.isNaN(Number(value))`. A Regex here (IMO) just seems overkill and is always harder to read.